app/controllers/shopping_carts_controller.rb

```diff
@@ -2,11 +2,11 @@ class ShoppingCartsController < ApplicationController
   before_action :get_shopping_cart
 
   def checkout
-
+    @cart = ShoppingCart.find(session[:cart_id])
   end
 
   def show
-    if @cart.total === 0
+    if @cart.shopping_cart_items.empty?
```

Comment:
This is good, but you can shorten it even more: `@cart.items?` I think. 
A tip: if you are in your rails console and have the `@cart` object available. Try to run `@cart.methods` to see what methods are available.